### PR TITLE
immich: Add timezone, model preloading environment variables

### PR DESCRIPTION
**Pull request by rouke-broersma**
Also removes some defaults such as port numbers, removes some non-default port numbers and fixes some deprecated environment variables.

Also fixes labels in accordance with #32

### CODE REVIEW

**Comment by rouke-broersma** [comment]
Needs to be tested

File: charts/immich/templates/server/deployment.yaml [hunk 1]
```diff
@@ -3,40 +3,31 @@ apiVersion: apps/v1
...
         - name: server
           image: "{{ .Values.server.image.registry }}/{{ .Values.server.image.repository }}:{{ .Values.server.image.tag }}"
           imagePullPolicy: {{ .Values.server.image.pullPolicy }}
           env:
+            - name: TZ
+              value: {{ .Values.timezone }}
             - name: IMMICH_TELEMETRY_INCLUDE
-              value: "all"
-            - name: IMMICH_METRICS_PORT
-              value: "8081"
```
Comment (rouke-broersma) [inline charts/immich/templates/server/deployment.yaml:37]: This env variable was replaced when the api and microservices images were merged and this was the default port number anyway

File: charts/immich/templates/server/deployment.yaml [hunk 2]
```diff
@@ -3,40 +3,31 @@ apiVersion: apps/v1
...
-            - name: IMMICH_PORT
-              value: "3001"
```
Comment (rouke-broersma) [inline charts/immich/templates/server/deployment.yaml:39]: This is a non-default port, replaced with the default port

File: charts/immich/templates/server/deployment.yaml [hunk 3]
```diff
@@ -3,40 +3,31 @@ apiVersion: apps/v1
...
-    app.kubernetes.io/name: {{ template "common.name" . }}
-    app.kubernetes.io/instance: {{ .Release.Name }}
-    app.kubernetes.io/component: "immich-server"
+    app.kubernetes.io/component: "server"
     {{- include "common.labels" . | nindent 4 }}
 spec:
   revisionHistoryLimit: 0
   replicas: {{ .Values.server.replicas }}
-  strategy:
-    type: Recreate
```
Comment (rouke-broersma) [inline charts/immich/templates/server/deployment.yaml:14]: Why did we use recreate? Immich says they support multiple replicas. Does this not work properly?

File: charts/immich/templates/machine-learning/deployment.yaml [hunk 4]
```diff
@@ -3,37 +3,42 @@ apiVersion: apps/v1
...
     spec:
       containers:
         - name: server
           image: "{{ .Values.machineLearning.image.registry }}/{{ .Values.machineLearning.image.repository }}:{{ .Values.machineLearning.image.tag }}"
           imagePullPolicy: {{ .Values.machineLearning.image.pullPolicy }}
           env:
-            - name: IMMICH_PORT
-              value: "3003"
-            - name: TRANSFORMERS_CACHE
-              value: /cache
```
Comment (rouke-broersma) [inline charts/immich/templates/machine-learning/deployment.yaml:36]: Env variable does not exist according to docs: https://immich.app/docs/install/environment-variables#machine-learning

Could replace with MACHINE_LEARNING_CACHE_FOLDER but /cache is the default anyway

File: charts/immich/templates/machine-learning/deployment.yaml [hunk 5]
```diff
@@ -3,37 +3,42 @@ apiVersion: apps/v1
...
+        app.kubernetes.io/component: "machine-learning"
+        {{- include "common.selectorLabels" . | nindent 8 }}
     spec:
       containers:
         - name: server
           image: "{{ .Values.machineLearning.image.registry }}/{{ .Values.machineLearning.image.repository }}:{{ .Values.machineLearning.image.tag }}"
           imagePullPolicy: {{ .Values.machineLearning.image.pullPolicy }}
           env:
-            - name: IMMICH_PORT
-              value: "3003"
```
Comment (rouke-broersma) [inline charts/immich/templates/machine-learning/deployment.yaml:34]: Default port so no need to set it

File: charts/immich/templates/machine-learning/deployment.yaml [hunk 6]
```diff
@@ -3,37 +3,42 @@ apiVersion: apps/v1
...
-            - name: TRANSFORMERS_CACHE
-              value: /cache
+            - name: TZ
+              value: {{ .Values.timezone }}
```
Comment (rouke-broersma) [inline charts/immich/templates/machine-learning/deployment.yaml:27]: Changes timezone when during import photo's don't contain a timezone in exif data. This was set to UTC before which is likely to be wrong. More likely to be our local timezone.

File: charts/immich/templates/machine-learning/deployment.yaml (hunk 5 above)
Comment (rouke-broersma) [reply charts/immich/templates/machine-learning/deployment.yaml]: Looks like immich does not properly use their own defaults..

File: charts/immich/templates/machine-learning/deployment.yaml [hunk 7]
```diff
@@ -3,37 +3,44 @@ apiVersion: apps/v1
...
   name: {{ template "common.fullname" . }}-machine-learning
   labels:
-    app.kubernetes.io/name: {{ template "common.name" . }}
-    app.kubernetes.io/instance: {{ .Release.Name }}
-    app.kubernetes.io/component: "immich-machine-learning"
+    app.kubernetes.io/component: "machine-learning"
     {{- include "common.labels" . | nindent 4 }}
 spec:
   revisionHistoryLimit: 0
   replicas: {{ .Values.server.replicas }}
```
Comment (FlorisFeddema) [inline charts/immich/templates/machine-learning/deployment.yaml:10]: Shouldnt this be .Values.machinelearning.replicas?

Comment (FlorisFeddema) [reply charts/immich/templates/machine-learning/deployment.yaml:10]: You didnt change it but we might as well clean it up :)

File: charts/immich/templates/server/deployment.yaml (hunk 3 above)
Comment (FlorisFeddema) [reply charts/immich/templates/server/deployment.yaml:14]: Didn't expect them to support multiple replicas and didn't look into it tbh 👍🏼

File: charts/immich/values.yaml [hunk 8]
```diff
@@ -47,6 +49,10 @@ machineLearning:
     repository: immich-app/immich-machine-learning
     tag: v1.122.2
     pullPolicy: IfNotPresent
+  
+  modelPreload:
+    objectRecognition: ''
```
Comment (FlorisFeddema) [inline charts/immich/values.yaml:54]: What would be valid values for this field?

**Review by FlorisFeddema — DISMISSED** [review]

File: charts/immich/templates/machine-learning/deployment.yaml (hunk 7 above)
Comment (rouke-broersma) [reply charts/immich/templates/machine-learning/deployment.yaml:10]: 🤦‍♀️

**Pull request merged**